src/lxc/tools/lxc_unshare.c

```diff
@@ -246,8 +246,13 @@ static int do_start(void *arg)
 		}
 	}
 
-	if ((start_arg->flags & CLONE_NEWNS) && start_arg->want_default_mounts)
-		lxc_setup_fs();
+	if (start_arg->flags & CLONE_NEWNS) {
```

Comment:
pre-morning-coffee question: does this mean that mounts done in the -attach will not be reflected in the container and vice versa?

Reply:
Yes. So right now we seem to have the problem that when you run `lxc-unshare` it can end up unmounting filesystems and then it's reflected on the host. Most notably `proc`. If this is ok then I can drop that change!